File: .agents/skills/create-contract-typescript/evals/bad/implementation_in_contract.ts
// BAD: Contract contains implementation
export interface IFileSystemPort {
    readFile(path: FilePath): Promise<FileContent>;
}

class FileAdapter implements IFileSystemPort {
    async readFile(path: FilePath): Promise<FileContent> {
        return fs.readFileSync(path.value()); // BAD: implementation in contract
    }
}

File: .agents/skills/create-contract-rust/evals/bad/implementation_in_contract.rs
// BAD: Contract contains implementation
pub trait IFileSystemPort: Send + Sync {
    async fn read_file(&self, path: &FilePath) -> Result<FileContent, FileReadError>;
}

impl IFileSystemPort for FileAdapter {
    async fn read_file(&self, path: &FilePath) -> Result<FileContent, FileReadError> {
        todo!() // BAD: implementation belongs in capabilities
    }
}
